crypto.go

```diff
@@ -66,5 +87,31 @@ func GenerateKey() ([]byte, error) {
 
 // EcRecover recovers the public key from a message, signature pair.
 func EcRecover(msg, signature []byte) ([]byte, error) {
-	return secp256k1.RecoverPubkey(msg, signature)
+	if len(signature) == 65 {
+		// parse the compact signature into its (`r`,`s`) scalars and `v` recovery bit
+		r, s, v, err := secpEc.ParseCompactRecoverableSignature(signature)
+		if err != nil {
+			return nil, err
+		}
+
+		pk, err := secpEc.RecoverPublicKey(msg, r, s, v)
+		if err != nil {
+			return nil, err
+		}
+
+		return pk.Bytes(), nil
+	}
+
+	// parse the compact signature into its (`r`,`s`) scalars (without `v` recovery bit)
```

Comment:
I feel like we probably want to handle other lengths explicitly?

Reply:
Wrt to just `EcRecover` or all methods that check signature lengths?

If all methods, were different sig lengths (other than 64, 65 bytes) handled prior to this refactor? 
- If yes, then i agree. 
- If no, then i would say handling other types of signatures should be a feature-add in a separate PR.